Recently used location in LBL modal .

sagarpreet-chadha opened the pull request:
Continued from https://github.com/publiclab/leaflet-blurred-location-display/issues/64#issuecomment-482655146 😄 !

Fixes https://github.com/publiclab/leaflet-blurred-location-display/issues/73 .

This PR does following : 

- [x]  Adds recently used locations in LBL modal . On clicking those badges , the map will pan to those coordinates  (hence faster use of LBL) .

- [x] The map by default will be at the last used location by the user . 

### Screenshots : 

<img width="910" alt="Screenshot 2019-04-18 at 8 24 05 PM" src="https://user-images.githubusercontent.com/14952645/56371234-5f5b4700-621a-11e9-965d-f10784925af9.png">

### GIF : 

![lbl_recent_location_gif](https://user-images.githubusercontent.com/14952645/56371290-7c901580-621a-11e9-80b3-c19c53e2fc62.gif)

jywarren commented:
This looks awesome!!!!!

I was wondering... is there another Bootstrap UI component we could use that would look a little less crowded? 

Could `place:lima-peru` and its coords become: `lima-peru (42.5, 35.3)` using a button with a badge maybe?

https://getbootstrap.com/docs/3.3/components/#badges

with the coords in the badge (maybe a `btn-default`, too?)?

![image](https://user-images.githubusercontent.com/24359/56377187-d02d3f80-61d7-11e9-9845-15c564de1097.png)

sagarpreet-chadha commented:
Okay, so i tried using a button and they are not fitting smoothly in the modal : 

<img width="739" alt="Screenshot 2019-04-20 at 1 13 00 PM" src="https://user-images.githubusercontent.com/14952645/56454619-2b466a00-6371-11e9-8456-dc662fba81af.png">



Finally, i tried a badge within a badge which looks more clean to me --- tell me what you think ?

<img width="728" alt="Screenshot 2019-04-20 at 1 32 43 PM" src="https://user-images.githubusercontent.com/14952645/56454621-33060e80-6371-11e9-81af-9bd3fff615fd.png">

sagarpreet-chadha commented:
@jywarren , we missed this PR 😄 . Kindly review --- How does the UI looks now ?

jywarren commented:
Sorry i took so long!!!! Missed this!!! Hmm, this looks good, but still not sure why it's not appearing as in the [screenshot above](https://github.com/publiclab/plots2/pull/5522#issuecomment-484587872). Would checking that it works in Bootstrap 4 help us debug?

sagarpreet-chadha commented:
<img width="814" alt="Screenshot 2019-06-27 at 5 45 43 PM" src="https://user-images.githubusercontent.com/14952645/60265669-e8ba8400-9903-11e9-8883-7b4c421200fe.png">



<img width="600" alt="Screenshot 2019-06-27 at 5 45 51 PM" src="https://user-images.githubusercontent.com/14952645/60265671-e9531a80-9903-11e9-8b15-450c28e2e91e.png">


@jywarren ...how does this looks now ?

jywarren commented on app/views/locations/_form.html.erb:
```diff
@@ -8,8 +8,34 @@
+          <% end %>
+          <a href="#" onclick="blurredLocation.goTo(<%= location[0].name.split(":")[1] %> , <%= location[1].name.split(":")[1] %>  , 6);"> 
+            <span class="badge badge-info" style="background-color: #337ab7 ; margin-left: 7px ;margin-top: 2px; ">
+              <p>
```
I tried removing the padding here!

jywarren commented on test/system/screenshots_test.rb:
```diff
@@ -69,4 +69,13 @@ class ScreenshotsTest < ApplicationSystemTestCase
     take_screenshot
   end
+  
+  test 'blog page with location modal' do
```
and I hope this works - added an automated screenshot of the modal! Let's see!

sagarpreet-chadha replied:
`Capybara::ElementNotFound: Unable to find css "a.blurred-location-input"`

sagarpreet-chadha replied:
We are not logged in and hence it makes sense that tests are not passing !

https://camo.githubusercontent.com/721dd60bb7945ee8c9212ddef02703969d961e50/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d6661696c757265735f746573745f626c6f675f706167655f776974685f6c6f636174696f6e5f6d6f64616c2e706e67

sagarpreet-chadha commented on test/system/screenshots_test.rb:
```diff
@@ -69,4 +69,13 @@ class ScreenshotsTest < ApplicationSystemTestCase
   end
+  
+  test 'blog page with location modal' do
+    visit '/people'
```
@jywarren ...i have changed the URL to people's page . The screenshot will not show the **recent locations** because we are not logged-in .

jywarren replied:
oh ok let me change it back with the log in code too. Let's see!

jywarren commented:
Although whoa, why are we on people page? Let me check...

jywarren commented:
Uh, hmm.

jywarren commented:
Hmm. @sagarpreet-chadha can you try rebasing this?

sagarpreet-chadha commented:
Okay tests are passing now . In the screenshot is makes sense that recent used location badges are not shown because there are none . 
@jywarren ...should we merge this ?

sagarpreet-chadha commented:
Okay do not merge yet , tests are not correct!

sagarpreet-chadha commented:
@jywarren ...i have rebased the PR . Kindly add the login sequence again in the screenshot test , i have removed the last commit by mistake . Apologies 😄 .

jywarren commented:
Ok! Looks like we have to log in as a different user to see recently used
locations?

On Fri, Jun 28, 2019, 2:32 PM plotsbot <notifications@github.com> wrote:

> 1 Warning
> ⚠️ There was an error with Danger bot’s Junit parsing: No JUnit file was
> found at output.xml
> 3 Messages
> 📖 @sagarpreet-chadha <https://github.com/sagarpreet-chadha> Thank you
> for your pull request! I’m here to help with some tips and recommendations.
> Please take a look at the list provided and help us review and accept your
> contribution! And *don’t be discouraged if you see errors* – we’re here
> to help.
> 📖 This pull request doesn’t link to a issue number. Please refer to the
> issue it fixes (if any) in the body of your PR, in the format: Fixes #123.
> 📖 # Screenshots 📸 (click to expand)5522-test_viewing_question_post.png
>
>
> <https://camo.githubusercontent.com/bbf6e76cb0636f639cbf10b132ac35d1612f8713/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f76696577696e675f7175657374696f6e5f706f73742e706e67>
> 5522-test_wiki.png
>
>
> <https://camo.githubusercontent.com/9ab5f3a14ebbe1231d18029306dcf6c9e8205719/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f77696b692e706e67>
> 5522-test_tag_page.png
>
>
> <https://camo.githubusercontent.com/cb56cb513596767d697ff5f176545d513eaa345e/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f7461675f706167652e706e67>
> 5522-test_blog_page_with_location_modal.png
>
>
> <https://camo.githubusercontent.com/9f10d5de89caaa53c2216d5e5a3755ab8f11e5bf/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f626c6f675f706167655f776974685f6c6f636174696f6e5f6d6f64616c2e706e67>
> 5522-test_login.png
>
>
> <https://camo.githubusercontent.com/b9b4aae8471d5b84b07c17b0b614288589fc2a3a/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f6c6f67696e2e706e67>
> 5522-test_wiki_page_with_inline_grids.png
>
>
> <https://camo.githubusercontent.com/5597703625a52d831a8ff28fb9d4a42bad35794c/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f77696b695f706167655f776974685f696e6c696e655f67726964732e706e67>
> 5522-test_questions.png
>
>
> <https://camo.githubusercontent.com/da62c48e3535f26363dc13772b9a58ff0168fce1/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f7175657374696f6e732e706e67>
> 5522-test_stats.png
>
>
> <https://camo.githubusercontent.com/de764d8a5303794c9f6dbffe799502b8ef8f291d/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f73746174732e706e67>
> 5522-test_tags.png
>
>
> <https://camo.githubusercontent.com/f31207480f5d7f87ad6ecd9638e8035f7fd74f38/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f746167732e706e67>
> 5522-test_people.png
>
>
> <https://camo.githubusercontent.com/99c157988193f8c20b8f5b3727acb084667a3af6/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f70656f706c652e706e67>
> 5522-test_front.png
>
>
> <https://camo.githubusercontent.com/2b262471d363e38a50db09baedde643017c31f72/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f66726f6e742e706e67>
> 5522-test_signup.png
>
>
> <https://camo.githubusercontent.com/11886a3415d3a720f2c1527991d0ba5c080b70f1/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f7369676e75702e706e67>
> 5522-test_questions_shadow.png
>
>
> <https://camo.githubusercontent.com/a2e01b11bf058e1fd83b3d17e3b8c542410b4867/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f7175657374696f6e735f736861646f772e706e67>
> 5522-test_blog.png
>
>
> <https://camo.githubusercontent.com/f6945526d191dbb29d6c5de06486d0cfcc461178/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f626c6f672e706e67>
> 5522-test_front_page_with_navbar_search_autocomplete.png
>
>
> <https://camo.githubusercontent.com/00a9bedbe7fb6c1e32faeb97f7b83d52290a02ce/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f66726f6e745f706167655f776974685f6e61766261725f7365617263685f6175746f636f6d706c6574652e706e67>
> 5522-test_viewing_the_dashboard.png
>
>
> <https://camo.githubusercontent.com/fc2e6a1f00c36331b015965478bd5a253bb06ec2/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f76696577696e675f7468655f64617368626f6172642e706e67>
>
> Learn about automated screenshots
> <https://github.com/publiclab/plots2/issues/5316>
>
> Generated by 🚫 Danger <https://danger.systems/>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/5522?email_source=notifications&email_token=AAAF6J3N76RRR26ETM6AOXTP4ZKILA5CNFSM4HG57YMKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODY23BKQ#issuecomment-506835114>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAF6JYJYCHOHTZEZZGWTFDP4ZKILANCNFSM4HG57YMA>
> .
>

jywarren commented:
Timing issue, i think. Restarting!

jywarren commented:
Hooray! 

![](https://camo.githubusercontent.com/9f10d5de89caaa53c2216d5e5a3755ab8f11e5bf/68747470733a2f2f73746f726167652e676f6f676c65617069732e636f6d2f706c6f7473322d73637265656e73686f74732f353532322d746573745f626c6f675f706167655f776974685f6c6f636174696f6e5f6d6f64616c2e706e67)

jywarren commented:
OK this looks good now!!!

![image](https://user-images.githubusercontent.com/24359/60368144-0f3cf580-99be-11e9-8a5d-b3a60ed8e2ed.png)

Merging once the last tests pass!

jywarren commented:
Awesome!!!

sagarpreet-chadha commented:
Awesome 😍 !!!
